lib/plaid/models.rb

```diff
@@ -47,19 +47,19 @@ class Error < BaseModel
       # Public: A user-friendly representation of the error message. nil if the
       # error is not related to user action.
       property :display_message
+
+      ##
+      # :attr_reader:
+      # Public: A list of reasons explaining why the error happened.
+      coerce_key :causes, ->(causes) { causes.map { |c| Cause.new(c) } }
```

Comment:
`property :causes, Array[Cause]` ?

Reply:
Doing that causes a circular dependency (`Error` depends on `Cause` which extends `Error`...). This is the recommended way to fix that.

Reply:
I see. But do you really need a recursive data structure here? (since `Cause` is inherited from `Error`, it can also have its own causes: `error.causes[0].causes[0]`, etc.). If yes, it's OK. If not, these classes can be remodeled as:

```ruby
class BaseError < BaseModel
  property :error_type
  property :error_code
  property :error_message
  property :display_message
end

class Cause < BaseError
  property :item_id
end

class Error < BaseError
  property :causes, coerce: Array[Cause]
end
```

Reply:
⚡️